catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/alias/AttributeAliasesImpl.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.alias;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AttributeAliasesImpl implements AttributeAliases {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AttributeAliasesImpl.class);
+
+  private ImmutableMap<String, String> aliases = ImmutableMap.of();
```

Comment:
Since we are encapsulating the map and protecting against puts, is it really necessary to make this an immutable map?

Reply:
If I didn't make it an ImmutableMap, values could be put by doing `getAliasMap().put("k", "v")`.  If you are wondering why I made the interface method `getAliasMap()` return `ImmutableMap` directly instead of `Map`, see https://stackoverflow.com/questions/38087900/is-it-better-to-return-an-immutablemap-or-a-map/38088079#38088079